corehq/apps/data_interfaces/tasks.py

```diff
@@ -130,25 +130,24 @@ def run_case_update_rules_for_domain(domain, now=None):
 
     for case_type, rules in rules_by_case_type.iteritems():
         boundary_date = AutomaticUpdateRule.get_boundary_date(rules, now)
-        case_id_chunks = AutomaticUpdateRule.get_case_ids(domain, case_type, boundary_date)
-
-        for case_ids in case_id_chunks:
-            for case in CaseAccessors(domain).iter_cases(case_ids):
-                migration_in_progress, last_migration_check_time = check_data_migration_in_progress(domain,
-                    last_migration_check_time)
-
-                time_elapsed = datetime.utcnow() - start_run
-                if (
-                    time_elapsed.seconds > HALT_AFTER or
-                    case_update_result.total_updates >= settings.MAX_RULE_UPDATES_IN_ONE_RUN or
-                    migration_in_progress
-                ):
-                    run_record.done(DomainCaseRuleRun.STATUS_HALTED, cases_checked, case_update_result)
-                    notify_error("Halting rule run for domain %s." % domain)
-                    return
-
-                case_update_result.add_result(run_rules_for_case(case, rules, now))
-                cases_checked += 1
+        case_ids = list(AutomaticUpdateRule.get_case_ids(domain, case_type, boundary_date))
```

Comment:
it depends on the rule, but probably less than a million, which i've been able to do in management commands before

i think the chunked approach broke down because it required a connection to stay open for the entire time the processing was happening. i'm going to give this a try and see if it's any better. eventually we're probably going to have to split this process by db